### ISSUE TITLE
Breakpoint not being hit when attaching to debugger

### ISSUE THREAD

**Issue by nimehta2010** [PowerShell/vscode-powershell#2792]
### System Details
Windows10

System Details Output
```

```

### Issue Description

I am experiencing a problem as the breakpoints are not getting hit when attaching debugger to different pwsh window.

#### Expected Behaviour

--It should stop at breakpoint

#### Actual Behaviour

-- It just executed without halting

### Attached Logs

[vscode-powershell.log](https://github.com/PowerShell/vscode-powershell/files/4865724/vscode-powershell.log)
[StartEditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4865725/StartEditorServices.log)
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4865726/EditorServices.log)

**Comment by TylerLeonhardt** [comment]
@nimehta2010 I see you're using attach to process debugging... can you confirm that you are attaching to the right process?

**Comment by nimehta2010** [comment]
Yes I am attaching to correct process.
Nishant Mehta

**Comment by TylerLeonhardt** [comment]
@nimehta2010 can you please share your `launch.json`?

**Comment by nimehta2010** [comment]
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "PowerShell: Interactive Session",
            "type": "PowerShell",
            "request": "launch",
            "cwd": ""
        },
        {
            "name": ".NET Core Attach",
            "type": "coreclr",
            "request": "attach",
            "processId": "${command:pickProcess}"
        },
        {
            "name": "PowerShell: Attach to PowerShell Host Process",
            "type": "PowerShell",
            "request": "attach",
            "runspaceId": 1
        }
    ]
}

**Comment by IAmIlliest** [comment]
Just dropping into say I have this same issue quite often. It then often ends up getting to the point where the debugger can't even attach to the process (reloading the extensions, VS Code and Powershell doesn't even seem to fix that, tends to need a reboot).

**Comment by LeopoldLerch** [comment]
Same here, Breakpoint does not get hit. However, I am sure i got the right process as "Pausing" is working, when I hit "Pause" during the execution of the script-file, the debugger jumps directly into the current position of the code and I can even step through it. However, any breakpoint in that same file is not hit.

I also noticed, that after disconnection of the debugger, the integrated console-window still is bound. Subsequent Debug-Sessions are not working, it just does not attach anymore. Only possibility is to restart the session of the integrated powershell.

I am using Powershell 7 (x64) by default, however, also tried setting the default powershell to Windows Powershell (5) did not change that issue.

Interactive mode or launching the file directly works. (however, sometimes i need a specific powershell version, thats why I need the "attach" working too.

**Comment by IAmIlliest** [comment]
Yep, confirming the same symptoms as Leo above. Probably unrelated, but just in case it isn't, I'm using Conda environments.

For what it's worth, I'm doing the bits below in a condition where VS Code PS Extension is no longer able to attach to another host process (it times out). Using PS 7.1.3:

- I can attach to the other (first) PS Host process from a second (external to VS Code) PS instance.
- I can also still attach to the other (first) PS Host from the internal terminal in VS Code.
- Once attached, I can set breakpoints in the other (first) PS Host process for the script and see that Get-PSBreakpoints is showing them.
- I can then seemingly successfully enter to begin to debug the runspace that the script is executing in on the other (first) PS Host process.
- I then run the script, initiating it 'locally' on the other (first) PS Host process.
- In both cases, those breakpoints will be ignored.

As such, it seems like the issue is not just VS Code or the PS extension in VS Code, Something is happening that is making the debugging engine for PS 'non-responsive' to breakpoints, or else it isn't attaching properly, despite reporting that is?

Either that, or I'm missing something else....

**Comment by Noctran** [comment]
I experience the same problem and can agree with the above descriptions.
